scene/2d/joint_2d.cpp

```diff
@@ -289,11 +295,98 @@ real_t PinJoint2D::get_softness() const {
+	return motor_target_velocity;
+}
+
+void PinJoint2D::set_enable_motor(bool p_enable_motor) {
+	if (enable_motor == p_enable_motor) {
+		return;
+	}
+	enable_motor = p_enable_motor;
+	queue_redraw();
+	if (is_configured()) {
+		PhysicsServer2D::get_singleton()->pin_joint_set_flag(get_joint(), PhysicsServer2D::PIN_JOINT_FLAG_ENABLE_MOTOR, enable_motor);
+	}
+}
+bool PinJoint2D::get_enable_motor() const {
+	return enable_motor;
+}
+void PinJoint2D::set_enable_limits(bool p_enable_limits) {
+	if (enable_limits == p_enable_limits) {
+		return;
+	}
+	enable_limits = p_enable_limits;
+	queue_redraw();
+	if (is_configured()) {
+		PhysicsServer2D::get_singleton()->pin_joint_set_flag(get_joint(), PhysicsServer2D::PIN_JOINT_FLAG_USE_LIMIT, enable_limits);
+	}
+}
+bool PinJoint2D::get_enable_limits() const {
+	return enable_limits;
+}
+
 void PinJoint2D::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_softness", "softness"), &PinJoint2D::set_softness);
 	ClassDB::bind_method(D_METHOD("get_softness"), &PinJoint2D::get_softness);
+	ClassDB::bind_method(D_METHOD("set_angular_limit_lower", "angular_limit_lower"), &PinJoint2D::set_angular_limit_lower);
+	ClassDB::bind_method(D_METHOD("get_angular_limit_lower"), &PinJoint2D::get_angular_limit_lower);
+	ClassDB::bind_method(D_METHOD("set_angular_limit_upper", "angular_limit_upper"), &PinJoint2D::set_angular_limit_upper);
+	ClassDB::bind_method(D_METHOD("get_angular_limit_upper"), &PinJoint2D::get_angular_limit_upper);
+	ClassDB::bind_method(D_METHOD("set_motor_target_velocity", "motor_target_velocity"), &PinJoint2D::set_motor_target_velocity);
+	ClassDB::bind_method(D_METHOD("get_motor_target_velocity"), &PinJoint2D::get_motor_target_velocity);
+	ClassDB::bind_method(D_METHOD("set_enable_motor", "enabled"), &PinJoint2D::set_enable_motor);
+	ClassDB::bind_method(D_METHOD("get_enable_motor"), &PinJoint2D::get_enable_motor);
+	ClassDB::bind_method(D_METHOD("set_enable_limits", "enabled"), &PinJoint2D::set_enable_limits);
+	ClassDB::bind_method(D_METHOD("get_enable_limits"), &PinJoint2D::get_enable_limits);
 
 	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "softness", PROPERTY_HINT_RANGE, "0.00,16,0.01,exp"), "set_softness", "get_softness");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "angular_limit/enable"), "set_enable_limits", "get_enable_limits");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "angular_limit/lower", PROPERTY_HINT_RANGE, "-180,180,0.1,radians"), "set_angular_limit_lower", "get_angular_limit_lower");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "angular_limit/upper", PROPERTY_HINT_RANGE, "-180,180,0.1,radians"), "set_angular_limit_upper", "get_angular_limit_upper");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "motor/enable"), "set_enable_motor", "get_enable_motor");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "motor/target_velocity", PROPERTY_HINT_RANGE, "-200,200,0.01,or_greater,or_less,suffix:m/s"), "set_motor_target_velocity", "get_motor_target_velocity");
```

Comment:
```suggestion
	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "motor/target_velocity", PROPERTY_HINT_RANGE, U"-200,200,0.01,or_greater,or_less,radians,suffix:\u00B0/s"), "set_motor_target_velocity", "get_motor_target_velocity");
```
This is angular yes?